stdlib/public/core/KeyPath.swift

```diff
@@ -774,6 +774,12 @@ internal enum KeyPathComponent: Hashable {
         return false
       }
       if let arg1 = argument1, let arg2 = argument2 {
+        // To compare for equality, they must be the same type. Check that that
+        // by checking if they have the same witnesses. If they have different
+        // witnesses then they cannot ever be equal.
```

Comment:
The little witness table for key path equality is generated on-demand for each key path pattern and can't be used as a source of equality, since it'll potentially be different for every key path pattern that shows up in source.

Reply:
Makes sense, I hadn't had a time to look at it yet. Do we have a way to determine if the equality witness is actually safe to call?

Reply:
I was thinking about it, and I'm not sure. We can't necessarily use the generic arguments packed into the index either because the same subscript could be instantiated from different generic contexts, or from a mix of concrete and generic contexts. We might have to add new information to the key path arguments to fix this hole (and also come up with a way to interoperate with existing binaries as best we can…)

Reply:
Thanks Joe. I'll adjust my dreams of a quick, simple fix....